_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+Make sure to exclude **appsettings.json** from Git so that you don't accidentally commit your client secret. You can generate a **.gitignore** file by running the command `dotnet new gitignore`.
+
+Now you'll need to add the authentication library. Run the following command inside the **Okta.Blog.Chat** folder.
+
+```bash
+dotnet add package Okta.AspNetCore
+```
+
+Now modify **Startup.cs** to use the Okta authentication provider.
+
+Modify the method `ConfigureServices(IServiceCollection services)` to look like the code below. I've commented out a line that we will use later for authorization.
+
+```cs
+public void ConfigureServices(IServiceCollection services)
+{
+    var oktaMvcOptions = new OktaMvcOptions()
+    {
+        OktaDomain = Configuration["OktaSettings:OktaDomain"],
+        ClientId = Configuration["OktaSettings:ClientId"],
+        ClientSecret = Configuration["OktaSettings:ClientSecret"],
+        Scope = new List<string> { "openid", "profile", "email" },
+    };
+
+    services.AddAuthentication(options =>
+    {
+        options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+        options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+        options.DefaultChallengeScheme = OktaDefaults.MvcAuthenticationScheme;
+    })
+    .AddCookie()
+    .AddOktaMvc(oktaMvcOptions);
+
+    services.AddRazorPages()
+    .AddRazorPagesOptions(options =>
+    {
+        //options.Conventions.AuthorizePage("/Chat");
+    });
+    services.AddSignalR();
+}
+```
+
+Make sure to add imports.
+
+```cs
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Okta.AspNetCore;
+using Okta.Blog.Chat.Hubs;
+```
+
+This will add the authentication provider, set the page **Chat** as an authorized page, and add SignalR support.
+
+Next, modify the method `Configure(IApplicationBuilder app, IWebHostEnvironment env)` to look like the code below. I've commented out a line that we will use later during our SignalR setup.
+
+```cs
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+    if (env.IsDevelopment())
+    {
+        app.UseDeveloperExceptionPage();
+    }
+    else
+    {
+        app.UseExceptionHandler("/Error");
+        // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+        app.UseHsts();
+    }
+
+    app.UseHttpsRedirection();
+    app.UseStaticFiles();
+
+    app.UseRouting();
+
+    app.UseAuthentication();
+    app.UseAuthorization();
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.MapRazorPages();
+        //endpoints.MapHub<ChatHub>("/chathub");
+    });
+}
+```
+
+The key difference here is `app.UseAuthentication();`.
+
+Next, create a new folder called **Hubs** and a new class file in that folder called **ChatHub.cs** this will provide our chat back-end.
```

Comment:
```suggestion
Next, create a new folder called **Hubs** and a new class file in that folder called **ChatHub.cs**.  This will provide our chat backend.
```
fix run-on sentence
backend per style guide